Library/Homebrew/test/utils_spec.rb

```diff
@@ -0,0 +1,340 @@
+      ).to eq("test")
+    end
+
+    describe Shell do
+      subject { described_class }
+
+      describe "#shell_profile" do
+        before(:each) do
+          ENV["SHELL"] = shell
+        end
+
+        context "default" do
+          let(:shell) { "/bin/another_shell" }
+          its(:shell_profile) { is_expected.to eq("~/.bash_profile") }
+        end
+
+        context "Sh" do
+          let(:shell) { "/bin/sh" }
+          its(:shell_profile) { is_expected.to eq("~/.bash_profile") }
+        end
+
+        context "Bash" do
+          let(:shell) { "/bin/bash" }
+          its(:shell_profile) { is_expected.to eq("~/.bash_profile") }
+        end
+
+        context "Zsh" do
+          let(:shell) { "/bin/zsh" }
+          its(:shell_profile) { is_expected.to eq("~/.zshrc") }
+        end
+
+        context "Ksh" do
+          let(:shell) { "/bin/ksh" }
+          its(:shell_profile) { is_expected.to eq("~/.kshrc") }
+        end
+      end
+    end
+
+    describe "#popen_read" do
+      specify "without block" do
+        expect(Utils.popen_read("sh", "-c", "echo success").chomp).to eq("success")
+        expect($?).to be_a_success
+      end
+
+      specify "with block" do
+        expect(
+          Utils.popen_read("sh", "-c", "echo success") do |pipe|
+            pipe.read.chomp
+          end,
+        ).to eq("success")
+        expect($?).to be_a_success
+      end
+    end
+
+    describe "#popen_write" do
+      specify "with block" do
+        Utils.popen_write("grep", "-q", "success") do |pipe|
+          pipe.write("success\n")
+        end
+        expect($?).to be_a_success
+      end
+    end
+
+    specify "#pretty_duration" do
+      expect(pretty_duration(1)).to eq("1 second")
+      expect(pretty_duration(2.5)).to eq("2 seconds")
+      expect(pretty_duration(42)).to eq("42 seconds")
+      expect(pretty_duration(240)).to eq("4 minutes")
+      expect(pretty_duration(252.45)).to eq("4 minutes 12 seconds")
+    end
+
+    specify "#plural" do
+      expect(plural(1)).to eq("")
+      expect(plural(0)).to eq("s")
+      expect(plural(42)).to eq("s")
+      expect(plural(42, "")).to eq("")
+    end
+
+    specify "#disk_usage_readable" do
+      expect(disk_usage_readable(1)).to eq("1B")
+      expect(disk_usage_readable(1000)).to eq("1000B")
+      expect(disk_usage_readable(1024)).to eq("1K")
+      expect(disk_usage_readable(1025)).to eq("1K")
+      expect(disk_usage_readable(4_404_020)).to eq("4.2M")
+      expect(disk_usage_readable(4_509_715_660)).to eq("4.2G")
+    end
+
+    specify "#number_readable" do
+      expect(number_readable(1)).to eq("1")
+      expect(number_readable(1_000)).to eq("1,000")
+      expect(number_readable(1_000_000)).to eq("1,000,000")
+    end
+
+    specify "#truncate_text_to_approximate_size" do
+      glue = "\n[...snip...]\n" # hard-coded copy from truncate_text_to_approximate_size
+      n = 20
+      long_s = "x" * 40
+
+      s = truncate_text_to_approximate_size(long_s, n)
```

Comment:
> it's a little unclear what's going on

Well, I have no idea what this method is actually doing. This was purely copy-pasted.